File: backend/app/schemas/band_member.py
from pydantic import BaseModel, EmailStr
from datetime import datetime
from typing import Optional


class BandMemberCreate(BaseModel):
    name: str
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    role: Optional[str] = None  # Musical role: Guitarist, Vocalist, etc.
    is_admin: bool = False


class BandMemberUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    role: Optional[str] = None
    is_admin: Optional[bool] = None


class BandMemberResponse(BaseModel):
    id: int
    band_id: int
    user_id: Optional[int] = None
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    profile_picture: Optional[str] = None
    role: Optional[str] = None
    is_admin: bool
    is_active: bool
    joined_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/types/bandMember.ts
export interface BandMember {
  id: number;
  band_id: number;
  user_id?: number | null;
  name?: string | null;
  email?: string | null;
  phone?: string | null;
  profile_picture?: string | null;
  role?: string | null;
  is_admin: boolean;
  is_active: boolean;
  joined_at: string;
}

export interface BandMemberCreate {
  name: string;
  email?: string | null;
  phone?: string | null;
  role?: string | null;
  is_admin?: boolean;
}

export interface BandMemberUpdate {
  name?: string;
  email?: string | null;
  phone?: string | null;
  role?: string | null;
  is_admin?: boolean;
}